DLPX-72423 modules.dep.bin is being removed preventing kernel modules from being loaded

pzakha opened the pull request:
This is caused because we have multiple parallel invocations of depmod at the same time, which are triggered by connstat. The real bug is with depmod that is not properly handling concurrent calls. That said, we have no reasons to call depmod at every invocation of connstat. I believe it is only necessary when we install the connstat kernel modules for the first time (which is already taken care of during image creation), and so was likely added during the initial prototyping of the package.

## Testing
ab-pre-push: http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/4462/

prakashsurya reviewed (commented):
This likely doesn't apply for our appliance, but for correctness, I think we need to call `depmod` _somewhere_.. e.g. for the ZFS modules, we call it in the package hooks, but I don't think we do that for this package.. so I'd like to also update this change to call `depmod` via the package hooks, if you're open to doing that.

prakashsurya reviewed (approved):
With that said, I'll approve as-is; but it'd be good to clean this up, or open a bug since this package likely won't work properly when used outside of the Delphix appliance.

pzakha commented:
@prakashsurya I've added the depmod call. Although things will most likely work as other kernel packages will be calling depmod during the install, I've added it for correctness.

re-testing: http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/4464/

prakashsurya commented:
> Although things will most likely work as other kernel packages will be calling depmod during the install

Right, I agree it's not an issue for our appliance; but if we install this package in isolation (e.g. a standard Ubuntu system), it likely won't work.
